packages/loader/container-loader/src/test/connectionStateHandler.spec.ts

```diff
@@ -1139,6 +1143,48 @@ describe("ConnectionStateHandler Tests", () => {
 		},
 	);
 
+	it("test 'read' reconnect & races ", async () => {
+		connectionStateHandler = createHandler(
+			false, // connectedRaisedWhenCaughtUp,
+			true, // readClientsWaitForJoinSignal
+		);
+
+		connectionStateHandler.establishingConnection({ text: "initial connect" });
+		connectionStateHandler.receivedConnectEvent(connectionDetails);
+		connectionStateHandler_receivedJoinSignalEvent(connectionDetails);
+		assert.strictEqual(
+			connectionStateHandler.connectionState,
+			ConnectionState.Connected,
+			"Client 1 should be in connected state",
+		);
+
+		connectionStateHandler.receivedDisconnectEvent({ text: "disconnect" });
+
+		connectionDetails2.mode = "read";
+		connectionStateHandler.establishingConnection({ text: "initial connect" });
+		connectionStateHandler.receivedConnectEvent(connectionDetails2);
+
+		// Clear Audience
+		connectionStateHandler_receivedLeaveSignalEvent(connectionDetails.clientId);
+
+		// We connected to different front-end, it still things that first client is there!
+		connectionStateHandler_receivedJoinSignalEvent(connectionDetails);
```

Comment:
If we get this duplicate join but no prior leave signal, how does Audience react?  Swallow it?  Assert?

Reply:
Audience eats duplicate signals (or non-matching signals, like leave without join). But it validates that if we have duplicate join, then IClient payload is the same